src/material-experimental/mdc-tooltip/harness/tooltip-harness.spec.ts

```diff
@@ -0,0 +1,98 @@
+import {HarnessLoader} from '@angular/cdk-experimental/testing';
+import {TestbedHarnessEnvironment} from '@angular/cdk-experimental/testing/testbed';
+import {Component} from '@angular/core';
+import {ComponentFixture, TestBed} from '@angular/core/testing';
+import {MatTooltipModule} from '@angular/material/tooltip';
+import {NoopAnimationsModule} from '@angular/platform-browser/animations';
+import {MatTooltipModule as MatMdcTooltipModule} from '../index';
+import {MatTooltipHarness} from './tooltip-harness';
+import {MatTooltipHarness as MatMdcTooltipHarness} from './mdc-tooltip-harness';
+
+let fixture: ComponentFixture<TooltipHarnessTest>;
+let loader: HarnessLoader;
+let harness: typeof MatTooltipHarness;
+
+describe('MatTooltipHarness', () => {
+  describe('non-MDC-based', () => {
+    beforeEach(async () => {
+      await TestBed.configureTestingModule({
+        imports: [MatTooltipModule, NoopAnimationsModule],
+        declarations: [TooltipHarnessTest],
+      }).compileComponents();
+
+      fixture = TestBed.createComponent(TooltipHarnessTest);
+      fixture.detectChanges();
+      loader = TestbedHarnessEnvironment.loader(fixture);
+      harness = MatTooltipHarness;
+    });
+
+    runTests();
+  });
+
+  describe('MDC-based', () => {
+    beforeEach(async () => {
+      await TestBed.configureTestingModule({
+        imports: [MatMdcTooltipModule, NoopAnimationsModule],
+        declarations: [TooltipHarnessTest],
+      }).compileComponents();
+
+      fixture = TestBed.createComponent(TooltipHarnessTest);
+      fixture.detectChanges();
+      loader = TestbedHarnessEnvironment.loader(fixture);
+      // Public APIs are the same as MatTooltipHarness, but cast
+      // is necessary because of different private fields.
+      harness = MatMdcTooltipHarness as any;
+    });
+
+    // TODO: enable after MDC tooltip is implemented
+    // runTests();
+  });
+});
+
+/** Shared tests to run on both the original and MDC-based tooltip. */
+function runTests() {
+  it('should load all tooltip harnesses', async () => {
+    const tooltips = await loader.getAllHarnesses(harness);
+    expect(tooltips.length).toBe(2);
+  });
+
+  it('should be able to open a tooltip', async () => {
+    const tooltip = await loader.getHarness(harness.with({id: 'one'}));
+    expect(await tooltip.isOpen()).toBe(false);
+    await tooltip.open();
+    expect(await tooltip.isOpen()).toBe(true);
+  });
+
+  // TODO(crisbeto): this fails because we have a setTimeout inside the tooltip which
+  // can't be flushed. Talk to Miles about how we should approach this in harnesses.
+  // it('should be able to close a tooltip', async () => {
```

Comment:
@mmalerba In the tooltip's `hide` method we do something like `setTimeout(/* some close logic */, 0)` which I wasn't able to flush during the unit tests. I tried `await`ing a `setTimeout` as well, but it didn't help. I could probably get it to work if I use `fakeAsync` with a `tick` for the test, but that means that whoever is consuming the harness would have to know to use it as well. How should we handle this case? I'm guessing that we don't to lock ourselves into using `fakeAsync` or `async` around some tests.

Reply:
When you say `await`ing a `setTimeout`, what did that code look like?

Reply:
I had something like this:

```ts
function wait(time: number) {
    return new Promise(resolve => {
        setTimeout(resolve, time);
    });
}
```

Then I had `await wait(20000)` right before the assertions, but the timer still hadn't fired.